### PR TITLE
Print input prior to running function runner 

### PR DESCRIPTION
**Pull request by mssalemi**
### Issue
https://github.com/Shopify/shopify-functions/issues/242

## What is this doing?

This PR adds input to `function-runner`'s output.  This will allow partners to confirm the input that was used to run the function. 

The update was adding input as a `String` type to the `FunctionRunResult`, and updated the `fmt` method so that it also displays the input along with the current output. 

Also, when we create the `FunctionRunResult` in engine - we initializes it with the input, which needs to be converted from a `Vec<u8>`. 

I am fairly unfamiliar with Rust, so please let me know if there is any improvements that can be made.

Screenshot 2024-07-15 at 3 03 46 PM

### CODE REVIEW

File: src/function_run_result.rs [hunk 1]
```diff
@@ -23,18 +23,21 @@ pub struct FunctionRunResult {
...
     pub logs: String,
+    pub input: String,
     pub output: FunctionOutput,
     #[serde(skip)]
     pub profile: Option<String>,
 }
 
 impl FunctionRunResult {
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
```
Comment (mssalemi) [inline src/function_run_result.rs:34]: Added this due to a `= note: `-D clippy::too-many-arguments` implied by `-D warnings`. Is this okay to do?

File: src/engine.rs [hunk 2]
```diff
@@ -223,6 +223,7 @@ pub fn run(params: FunctionRunParams) -> Result<FunctionRunResult> {
         memory_usage,
         instructions,
         logs.to_string(),
+        String::from_utf8(input).unwrap(),
```
Comment (DuncanUszkay1) [inline src/engine.rs:226]: Shouldn't ship this with an unwrap here- should use the `?` syntax: https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html

That paired with our use of [anyhow](https://docs.rs/anyhow/latest/anyhow/) will ensure the errors raised by this are included as part of the method's result

File: src/function_run_result.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DuncanUszkay1) [reply src/function_run_result.rs:34]: Seems like we should be able to use regular struct initialization here without any additional boilerplate, unless I'm missing something

**Comment by mssalemi** [comment]
> Left two comments to be addressed- happy to pair on them and warmup my Rustfu 😄

Down! Free I'm Wednesday if you are, just getting caught up on script-editor stuff today (and off tomorrow)

**Comment by mssalemi** [comment]
@nickwesselman Can you take a look if the display input to function-runnner is what we want. Seems straight forward but just wanted to check - Let me know if any changes / edits are needed :)

File: src/function_run_result.rs [hunk 3]
```diff
@@ -139,58 +127,83 @@ mod tests {
...
+        let input_vec = serde_json::to_vec(&input_json).unwrap();
+        let mock_input_string = String::from_utf8(input_vec).unwrap();
+
         let function_run_result = FunctionRunResult {
             name: "test".to_string(),
             size: 100,
             memory_usage: 1000,
             instructions: 1000,
             logs: "test".to_string(),
+            input: mock_input_string.clone(),
```
[the remark this answers is not in the document]
Comment (DuncanUszkay1) [reply src/function_run_result.rs:170]: It occurs in another test below as well

**Comment by mssalemi** [comment]
> Should we cut a new version as well?

Yea believe so, just to make sure - would that be following this [guide](https://vault.shopify.io/page/Publish-a-new-version-of-an-internal-gem~dhbc57d.md)?

**Review by DuncanUszkay1 — APPROVED** [review]
Approved conditionally on cutting a new major version following the instructions linked

**Review by adampetro — APPROVED** [review]

**Comment by mssalemi** [comment]
> Approved conditionally on cutting a new major version following the instructions linked

Sounds good, I have the release [here](https://github.com/Shopify/function-runner/releases/edit/untagged-c5b242a39a951afc56ed). And will bump the version to `v.5.1.4` after this PR merges.

**Pull request merged**